src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1290,6 +1532,112 @@ public void Dispose()
             return context.LastDatabaseChangeVector;
         }
 
+        private static bool TryMergeIncrementResults(DocumentsOperationContext context, TimeSeriesSegmentHolder segmentHolder)
+        {
+            if (segmentHolder.ReadOnlySegment.NumberOfLiveEntries < 2)
```

Comment:
why? what if I delete a range that contains increment values as well?

Reply:
before actually deleting a range (in `DeleteTimestampRange`), we call `TryMergeIncrementResults` so if there are increment entries to merge we will merge it there.